docker/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Base Python (debian) image
-FROM python:3.11
+FROM python:3.12
```

Comment:
Is this strictly required? If not I suggest we leave it out of this PR.

Reply:
Reverted the change in the dockerfile.